CairoMakie/src/utils.jl

```diff
@@ -258,3 +261,12 @@ end
 function mesh_pattern_set_corner_color(pattern, id, c::Colorant)
     Cairo.mesh_pattern_set_corner_color_rgba(pattern, id, rgbatuple(c)...)
 end
+
+# not piracy
+function Cairo.CairoPattern(color::Makie.AbstractPattern)
```

Comment:
This should probably dispatch on ImagePattern instead, since if we add a specifically vector pattern it may be too broad.

Reply:
we can't
```julia
julia> p = Pattern('/');

julia> typeof(p)
Makie.LinePattern

julia> typeof(p) |> supertype
Makie.AbstractPattern{ColorTypes.RGBA{Float32}}
```

Reply:
if we add the other thing we can add a dispatch site for that, I don't think it matters much here

Reply:
the logic right now is we always convert it to a image because that's what `CairoImageSurface` consumes